modules/mcuboot/CMakeLists.txt

```diff
@@ -6,14 +6,33 @@ if(CONFIG_BOOTLOADER_MCUBOOT)
 
   include(${ZEPHYR_BASE}/../nrf/cmake/fw_zip.cmake)
 
-  function(sign to_sign_hex output_prefix offset sign_depends signed_hex_out)
-    set(op ${output_prefix})
-    set(signed_hex ${op}_signed.hex)
-    set(${signed_hex_out} ${signed_hex} PARENT_SCOPE)
-    set(to_sign_bin ${op}_to_sign.bin)
-    set(update_bin ${op}_update.bin)
-    set(moved_test_update_hex ${op}_moved_test_update.hex)
-    set(test_update_hex ${op}_test_update.hex)
+  function(sign)
+    # Signs a hex image
+    #
+    # Required arguments are:
+    # SIGNED_BIN_FILE_IN - (required) Hex image to sign
+    # SIGNED_HEX_FILE_NAME_PREFIX - (required) Prefix of the output image
+    # START_ADDRESS_OFFSET - (required) Offset
+    # SIGNED_HEX_FILE_OUT - (required) Signed hex output image
+    # DEPENDS - (optional) One or more dependencies for signing the hex image
+
+    set(oneValueArgs SIGNED_BIN_FILE_IN SIGNED_HEX_FILE_NAME_PREFIX START_ADDRESS_OFFSET SIGNED_HEX_FILE_OUT)
+    set(multiValueArgs DEPENDS)
+    cmake_parse_arguments(SIGN_ARG "" "${oneValueArgs}" "${multiValueArgs}" ${ARGN})
+
+    if (NOT DEFINED SIGN_ARG_SIGNED_BIN_FILE_IN
+        OR NOT DEFINED SIGN_ARG_SIGNED_HEX_FILE_OUT
+        OR NOT DEFINED SIGN_ARG_SIGNED_HEX_FILE_NAME_PREFIX
+        OR NOT DEFINED SIGN_ARG_START_ADDRESS_OFFSET)
+      message(FATAL_ERROR "Missing parameter, required: SIGNED_BIN_FILE_IN SIGNED_HEX_FILE_NAME_PREFIX START_ADDRESS_OFFSET SIGNED_HEX_FILE_OUT")
+    endif()
+
+    set(signed_hex ${SIGN_ARG_SIGNED_HEX_FILE_NAME_PREFIX}_signed.hex)
+    set(${SIGN_ARG_SIGNED_HEX_FILE_OUT} ${signed_hex} PARENT_SCOPE)
```

Comment:
Just wondering, why is it the function that decides the name and location of the signed file, and not the caller of the function that decides where and how the output image should be named ?